### PR TITLE
PHPStan/DOCBlock fixes ...

### PR DESCRIPTION
**Pull request by sreichel**
### Description (*)

- fixed "a few" phpstan errors
- added some DOCs
- removed useless `@return void`

### Related Pull Requests
#2146

### Manual testing scenarios (*)

1. run `vendor/bin/phpstan analyze -c .github/phpstan.neon`

### Contribution checklist (*)
 - [x] Pull request has a meaningful description of its purpose
 - [x] All commits are accompanied by meaningful commit messages
 - [x] All automated tests passed successfully (all builds are green)
 - [x] Add yourself to contributors list

### CODE REVIEW

**Review by fballiano — APPROVED** [review]

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php [hunk 1]
```diff
@@ -134,6 +134,7 @@ public function getRoot($parentNodeCategory=null, $recursionLevel=3)
      * If ids are empty, default tree with depth = 2 will be returned.
      *
      * @param array $ids
+     * @return mixed|Varien_Data_Tree_Node|null
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php:137]: Checking all the [`Mage::register('root', $root);`](https://github.com/OpenMage/magento-lts/search?q=Mage::register(%27root%27,%20$root);), I only see that `$root` can only be

```suggestion
     * @return Varien_Data_Tree_Node|null
```

No?

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Grid.php [hunk 2]
```diff
@@ -52,6 +54,10 @@ protected function _prepareCollection()
         return parent::_prepareCollection();
     }
 
+    /**
+     * @return void
```
Comment (kiatng) [inline app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Grid.php:58]: Just curious, why `@return void` is not useless here?

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php:137]: No! :)

PhpStan just knows about this ...

```
    /**
     * Retrieve a value from registry by a key
     *
     * @param string $key
     * @return mixed
     */
    public static function registry($key)
```

It does not know the content of `$root`.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Grid.php (hunk 2 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Grid.php:58]: This method has no return statement, but the parrent method has one.

(adding `return $this;` would be possible too - maybe in another PR)

Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Product/Attribute/Set/Grid.php:58]: Thanks, it makes sense now. I'm learning so much new stuff.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php (hunk 1 above)
Comment (kiatng) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php:137]: Tricky, I see it now. Thanks

**Review by kiatng — DISMISSED** [review]
Approved because PHPStan didn't complain.

**Comment by sreichel** [comment]
Last is PR ... for some reason comment lines did not work anymore ... moved them to excludePaths.

File: app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php (hunk 1 above)
Comment (sreichel) [reply app/code/core/Mage/Adminhtml/Block/Catalog/Category/Abstract.php:137]: @kiatng another examples is`@param Varien_Object|Same_Name` ... in most cases `Varien_Object` is to match a parent methods return type or typehints. `Same_Name` is used to look up for child classes and their methods.

**Review by kiatng — APPROVED** [review]

File: app/code/core/Mage/Catalog/Model/Product/Option/Type/File.php [hunk 3]
```diff
@@ -713,9 +713,8 @@ protected function _initFilesystem()
     /**
      * Create Writeable directory if it doesn't exist
      *
-     * @param string Absolute directory path
-     * @return void
-     */
+     * @param string $path Absolute directory path
+      */
```
Comment (luigifab) [inline app/code/core/Mage/Catalog/Model/Product/Option/Type/File.php:717]: Damned! A strange space!

**Review by luigifab — APPROVED** [review]
I suppose PHPStan says always the truth. Not tested.
Doc block changes, some `return` added, one undefined variable defined, 2x `->useProductItem()` changed to `->useProductItem(true)`.

**Review by ADDISON74 — APPROVED** [review]

File: app/code/core/Mage/Catalog/Model/Product/Option/Type/File.php (hunk 3 above)
Comment (sreichel) [reply app/code/core/Mage/Catalog/Model/Product/Option/Type/File.php:717]: Will be eliminated with next PR :eyes:

**Comment by sreichel** [comment]
> I suppose PHPStan says always the truth. Not tested. Doc block changes, some `return` added, one undefined variable defined, 2x `->useProductItem()` changed to `->useProductItem(true)`.

All changes are manually reviewed.

- added returns to nothing else than `return $this` ... (will be unified in later PR)
-  `useProductItem()` ... either this methot accecpts no paramenters or dault value is `true`.

**Pull request merged**